### ISSUE TITLE
The plugin does not take account global function

### ISSUE THREAD

**Issue by FuriouZz** [TypeStrong/fork-ts-checker-webpack-plugin#515]
## Current behavior
I created a transformer that replace `asset_filter("pattern")` into a Record<string, string>

For example

```ts
const Assets = asset_filter("projects/webm")
```
Result
```ts
const Assets = { 
  "assets/projects/webm/video1.webm": "assets/projects/webm/video1-a3edf576765dfgdfgdfg.webm",
  "assets/projects/webm/video2.webm": "assets/projects/webm/video2-e8dfd9asd85asfgfhjfd4.webm"
}
```

I have declared `asset_filter` as a global function to prevent the checker

**assets.d.ts**
```ts
declare global {
  export function asset_filter(key: string): Record<string, string>
}

// or
declare function asset_filter(key: string): Record<string, string>
```

The plugin triggers an error message :

```txt
Cannot find name 'asset_filter'.
```

## Expected behavior
Do not trigger an error

## Issue reproduction repository
https://github.com/FuriouZz/fork-ts-checker-webpack-plugin-issue-515

## Environment
 - **fork-ts-checker-webpack-plugin**: ^5.1.0
 - **typescript**: 4.0.2
 - **webpack**: ^4.44.1
 - **os**: Windows 10

**Comment by piotr-oles** [comment]
Is this working with `tsc --watch`?

**Comment by FuriouZz** [comment]
Yes, it worked.

I noticed the exact case where it does not work

**assets.d.ts**
```ts
import { PAGE } from "./PAGE.ts"
declare function asset_url(key: keyof typeof PAGE["assets"]): string
declare function asset_filter(key: string): Record<string, string>
```

**PAGE.ts**
```ts
export const PAGE = {
  "assets": {
    "webm/video1.webm": {
      "path": "/webm/video1-q2434fsd433655fgd.webm",
      "url": "/webm/video1-q2434fsd433655fgd.webm"
    },
    "webm/video2.webm": {
      "path": "/webm/video2-q2434fsd433655fgd.webm",
      "url": "/webm/video2-q2434fsd433655fgd.webm"
    },
    "mp4/video2.mp4": {
      "path": "/mp4/video2-q2434fsd433655fgd.mp4",
      "url": "/mp4/video2-q2434fsd433655fgd.mp4"
    },
  }
}
```

It seems to work when I do not import *PAGE.ts*.

This code compile

```ts
declare function asset_url(key: "webm/video1.webm"|"webm/video2.webm"|"mp4/video1.mp4"): string
declare function asset_filter(key: string): Record<string, string>
```

**Comment by FuriouZz** [comment]
I created a repository to reproduce https://github.com/FuriouZz/fork-ts-checker-webpack-plugin-issue-515

**Comment by piotr-oles** [comment]
Thanks!

**Comment by tangtangtangtangtang** [comment]
same problem. Will this problem be fixed in the next release?

**Comment by piotr-oles** [comment]
I didn't have time yet to look at this, I will take a look today :)

**Comment by piotr-oles** [comment]
Sorry for the delay, I've just tried your repro repository and it also doesn't work for `tsc --watch --noEmit`:
![image](https://user-images.githubusercontent.com/8374473/98293243-42c66180-1fae-11eb-99b3-b417a9425be2.png)

(maybe you did use globally installed `tsc` which points to a different version than locally installed typescript?)

**Comment by FuriouZz** [comment]
Hola 👋 No I only use installed `tsc` from `package.json`. I do not think to use the VSCode version.

**Comment by piotr-oles** [comment]
Hmmm... I don't know what is VSCode version - I did run `tsc --watch --noEmit` in the standard terminal using `tsc` from `node_modules` from your reproduction repository. Am I missing something?

**Comment by FuriouZz** [comment]
This is weird. This form seems to work

```ts
import { PAGE } from "./PAGE";
declare global {
  function asset_url(key: keyof typeof PAGE['assets']): string
  function asset_filter(key: string): Record<string, string>
}
```

In my early test, I used `declare global` and I got an error for days. Now it seems to work. I do not know =(. 

I will continue more test case.

**Comment by piotr-oles** [comment]
I'm closing this for now for maintenance reasons. Feel free to comment here if you find that something is still off with the plugin :)
